src/transformers/modeling_utils.py

```diff
@@ -2671,15 +2682,6 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
 
         if isinstance(device_map, str):
             special_dtypes = {}
-            if load_in_8bit:
-                special_dtypes.update(
-                    {
-                        name: torch_dtype
-                        for name, _ in model.named_parameters()
-                        if any(m in name for m in modules_to_not_convert)
-                    }
-                )
-
```

Comment:
I think this block is needed as it will be used to keep the modules that are inside `modules_to_not_convert` in the desired `dtype` - also we might need to change the condition to `if load_in_8bit or load_in_4_bit`

Reply:
Done!